packages/manager-react-components/src/components/datagrid/datagrid-topbar.component.tsx

```diff
@@ -80,28 +80,35 @@ export const DatagridTopbar = <T,>({
           >
             <div className="flex justify-end items-center">
               {isSearchable && (
-                <OdsInput
-                  isClearable
-                  onOdsClear={() => {
-                    search?.onSearch(null);
-                    search?.setSearchInput('');
-                  }}
-                  type={ODS_INPUT_TYPE.search}
-                  id="datagrid-searchbar"
-                  name="datagrid-searchbar"
-                  placeholder={search?.placeholder}
-                  defaultValue={search?.searchInput}
+                <form
                   className="mr-[5px]"
-                  data-testid="datagrid-searchbar"
-                  onOdsChange={(event) =>
-                    search?.setSearchInput(event?.detail?.value?.toString())
-                  }
-                  onKeyDown={(event) => {
-                    if (event.key === 'Enter')
-                      search?.onSearch(search?.searchInput);
+                  onSubmit={(e) => {
+                    e.preventDefault();
+                    search?.onSearch(search?.searchInput);
                   }}
-                  value={search?.searchInput}
-                />
+                >
+                  <OdsInput
+                    isClearable
+                    onOdsClear={() => {
+                      search?.onSearch(null);
```

Comment:
We could take the opportunity to fix the `onSearch` signature, because this is not correct: 
  
```ts
onSearch: (search: string | null) => void;
```

Reply:
Please check again, now it won't be ever called again with null, if it's empty then it will be an empty string instead, onSearch will never be called with undefined or null value, `search.searchInput` can't be undefined unless `search` it self is not defined according to the type of `SearchProps`